src/quantuminspire/qiskit/circuit_parser.py

```diff
@@ -47,6 +47,29 @@ def _gate_not_supported(_stream: StringIO, instruction: QasmQobjInstruction, _bi
         else:
             raise ApiError('Gate {} not supported'.format(instruction.name.lower()))
 
+    @staticmethod
+    def _cz(stream: StringIO, instruction: QasmQobjInstruction) -> None:
+        """ Translates the controlled Z element.
+
+        Args:
+            stream: The string-io stream to where the resulting cQASM is written.
+            instruction: The Qiskit instruction to translate to cQASM.
+
+        """
+        stream.write('CZ q[{0}], q[{1}]\n'.format(*instruction.qubits))
+
+    @staticmethod
+    def _c_cz(stream: StringIO, instruction: QasmQobjInstruction, binary_control: str) -> None:
+        """ Translates the binary-controlled controlled Z element.
+
+        Args:
+            stream: The string-io stream to where the resulting cQASM is written.
+            instruction: The Qiskit instruction to translate to cQASM.
+            binary_control: The multi-bits control string. The gate is executed when all specified classical bits are 1.
+
+        """
+        stream.write('C-CZ {0}q[{1}], q[{2}]\n'.format(binary_control, *instruction.qubits))
```

Comment:
Could you make this an f-string, and don't you miss a space between `{0}` and `q`?

Reply:
I'll do it for the whole file then. The space is in the binary_control string

Reply:
I think the 'old type' string is more clear than f string. Maybe you could look at method _cz and _cx and check yourself. There is no need to go to f-strings, is there?

Reply:
There is not necessarily, expect that Python prescribes the use of it in their style guide.